### PR TITLE
Add esp32-tester.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: tests/hw/esp-tester/mini-jag.toit [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  ids := containers.images.map: | image/containers.ContainerImage |
+    image.id
+  ids.do: | id/Uuid |
+    if id != containers.current:
+      containers.uninstall id
+
+  network := net.open
+  server-socket := network.tcp-listen 0
+  print "$network.address:$server-socket.local-address.port"
+  print "MINI-JAG LISTENING"
```
Comment (kasperl) [inline tests/hw/esp-tester/mini-jag.toit:31]: Maybe use the constant from shared.toit?

File: tests/hw/esp-tester/tester.toit [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+import host.directory
+import host.file
+import host.pipe
+import monitor
+import net
+import system
+import uart
+import .shared
+
+MINI-JAG-LISTENING ::= "MINI-JAG LISTENING"
```
Comment (kasperl) [inline tests/hw/esp-tester/tester.toit:17]: Use the constant from shared.toit?

File: tests/hw/esp-tester/tester.toit [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+TEST-STARTED ::= "TEST STARTED"
```
Comment (kasperl) [inline tests/hw/esp-tester/tester.toit:18]: Where is this used?

File: tests/hw/esp-tester/tester.toit [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+ALL-TEST-DONE ::= "All tests done"
```
Comment (kasperl) [inline tests/hw/esp-tester/tester.toit:19]:
```suggestion
ALL-TESTS-DONE ::= "All tests done"
```

File: tests/hw/esp-tester/tester.toit [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+      try:
+        reader := port.in
+        stdout := pipe.stdout
+        while data/ByteArray? := reader.read:
+          if not is-active: continue
+          stdout.write data
+          collected-output += data.to-string-non-throwing
+          if not ready-latch.has-value and collected-output.contains "\n$MINI-JAG-LISTENING":
+            ready-latch.set true
+          if not all-tests-done.has-value and collected-output.contains ALL-TEST-DONE:
```
Comment (kasperl) [inline tests/hw/esp-tester/tester.toit:151]:
```suggestion
          if not all-tests-done.has-value and collected-output.contains ALL-TESTS-DONE:
```

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tests/hw/esp-tester/mini-jag.toit (hunk 1 above)
Comment (floitsch) [reply tests/hw/esp-tester/mini-jag.toit]: done.

File: tests/hw/esp-tester/tester.toit (hunk 2 above)
Comment (floitsch) [reply tests/hw/esp-tester/tester.toit]: done.

File: tests/hw/esp-tester/tester.toit (hunk 4 above)
Comment (floitsch) [reply tests/hw/esp-tester/tester.toit]: done.

File: tests/hw/esp-tester/tester.toit (hunk 5 above)
Comment (floitsch) [reply tests/hw/esp-tester/tester.toit]: done.

File: tests/hw/esp-tester/tester.toit (hunk 4 above)
Comment (floitsch) [reply tests/hw/esp-tester/tester.toit]: done.

File: tests/hw/esp-tester/tester.toit (hunk 3 above)
Comment (floitsch) [reply tests/hw/esp-tester/tester.toit]: removed.
done.

**Pull request merged**